src/App.js

```diff
@@ -27,7 +26,6 @@ class App extends Component {
   render() {
     return (
       <React.Fragment>
-        <Sidebar />
```

Comment:
I moved this inside Navigation for the time being - just to show you how to work around the issues with state. But I still feel it doesn't belong there.

I think we could bring it back here, together with the SidebarButton and the related `useState()` thingie.
Ideally, we end up with something like this:

```
const [open, setOpen] = useState(true)

... render():

<Sidebar isOpen={open} /> // <= inside this Sidebar, I'd introduce a wrapper <div className={props.open ? "" : "msb-x"}>
<SidebarButton onClick={toggle open like we do}>
```

Reply:
I didn't wanted to play with passing props from child to its grandparent (from Button, through navigation to app.js) so I put this into the CustomerContext and now Sidebar is like this in App.js.

`class App extends Component {
  render() {
    return (
      <React.Fragment>
        <CustomerConsumer>
          {(ctx) => {
            return <Sidebar isSidebarOpen={ctx.sidebar} />;
          }}
        </CustomerConsumer>
        <Navigation />
        <Switch>
          <Route exact path="/" component={MainPage}></Route>
          <Route path="/products" component={ProductList}></Route>
          <Route path="/details/:productId" component={Details}></Route>
          <Route path="/cart" component={Cart}></Route>
          <Route path="/success" component={SuccessPage}></Route>
          <Route component={Default}></Route>
        </Switch>
        <Modal />
        <Footer />
        <ToastContainer {...toastOptions} />
      </React.Fragment>
    );
  }
}`